JSTask2.js

```diff
@@ -0,0 +1,293 @@
+{ name: "D", age: 18, college: "UVCE", stream: 'CSE', currentYear: 3 },
+{ name: "E", age: 22, college: "KEAM", stream: 'AEI', currentYear: 4 },
+{ name: "F", age: 21, college: "NETE", stream: 'EEE', currentYear: 3 }
+]
+a. There is election voting for the post of Chairperson, for which only final year students are eligible. Create a list satisfying the above condition.
+b. Student B seems to have the wrong college name in the above data. Return an above array with the college name replaced as “NETE”.
+c. Find the total number of students that would be eligible for placements this year.
+d. Create an array with students with stream ‘EEE' and in the order of their currentYear.
+e. Create a new array for the students above the age of 21, add a new property ‘cgpa’. */
+
+const students = [
+    {
+        name: "A",
+        age: 17,
+        college: "UVCE",
+        stream: 'CSE',
+        currentYear: 1
+    },
+    {
+        name: "B",
+        age: 19,
+        college: "KEAM",
+        stream:'ECE',
+        currentYear: 2
+    },
+    {
+        name: "C",
+        age: 23,
+        college: "UVCE",
+        stream: 'EEE',
+        currentYear: 4
+    },
+    {
+        name: "D",
+        age: 18,
+        college: "UVCE",
+        stream: 'CSE',
+        currentYear: 3
+    },
+    {
+        name: "E",
+        age: 22,
+        college: "KEAM",
+        stream: 'AEI',
+        currentYear: 4
+    },
+    {
+        name: "F",
+        age: 21,
+        college: "NETE",
+        stream: 'EEE',
+        currentYear: 3
+    }
+];
+
+// a. Only final students are eligible. Create a list satisfying the above conditions.
+console.log(students.filter(fy => fy.currentYear === 4));
+
+// b. B has wrong college. Replace the name of the college to NETE.
+let val = [];
+val = students.filter((na) => na.name === 'B');
+console.log(val);
+console.log(val.length);
+
+let change = students.filter((clg) => clg.name === 'B')[0].college;
+console.log(change);
+let ch = change.replace(/KEAM/g, 'NETE');     //using RegEx method
+console.log(ch);
+
+// c. Total number of students eligible for placements.
+console.log(students.filter((pla) => pla.currentYear === 4));
+let counter = 0;
+for(let i=0;i<students.length;i++)
+{
+    if(students[i].currentYear === 4)
+    {
+        counter++;
+    }
+}
+console.log(counter);
+
+// d. Array with students with stream 'EEE' & in order of current year.
+let array3 = [];
+for(let i=0;i<students.length;i++)
+{
+    if(students[i].stream === "EEE")
+    {
+        array3.push(students[i].name);
+    }
+}
+console.log(array3);
+
+// e. Array with students above age of 21 with a new property 'CGPA'.
+let array4 = [];
+for(let i=0;i<students.length;i++)
+{
+    if(students[i].age === 21)
+    {
+        array4.push(students[i].age);
+    }
+}
+console.log(array4);
+let a1 = [];
+a1 = students.filter(g => g.age === 21);
+console.log(a1);
+a1['CGPA'] = 10;
+console.log(a1);
+
+
```

Comment:
Not getting the expected output, it should include students with ages above 21, not 21 itself.  eg: {name: "E", age: 22, college: "KEAM", stream: 'AEI', currentYear: 4, CGPA: 10}